generator/src/main/java/org/wildfly/swarm/config/generator/generator/SubsystemPlan.java

```diff
@@ -65,9 +65,21 @@ void plan() {
             ModelNode attrs = each.getDescription().get(ATTRIBUTES);
             List<Property> props = attrs.asPropertyList();
             for (Property prop : props) {
+                ModelNode accessType = prop.getValue().get(ACCESS_TYPE);
+                if ( ! accessType.asString().contains( "write" ) ) {
+                    continue;
+                }
+
                 ModelNode deprecated = prop.getValue().get(DEPRECATED);
                 if (deprecated.isDefined()) {
-                    attrs.remove(prop.getName());
+                    String since = deprecated.get(SINCE).asString();
+                    if (since.startsWith("5.") || since.startsWith("4.")) {
+                        //System.err.println( "keeping: " + prop.getName() + " from " + each.getAddress() + " since " + deprecated.asString() );
+                        // keep it
```

Comment:
Could/should we maybe mark the deprecated-but-kept elements as `@Deprecated`? Not sure how much work that would be, just an idea. Maybe doesn't make a lot of sense, because we no longer promote the Java API for direct usage.

Reply:
Good idea, but not today.